azurerm/internal/services/attestation/tests/attestation_resource_test.go

```diff
@@ -0,0 +1,308 @@
+	}
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, &rawCert, &rawCert, &privateKey.PublicKey, privateKey)
+	if err != nil {
+		return "", fmt.Errorf("unable to create test certificate: %+v", err)
+	}
+
+	encoded := &bytes.Buffer{}
+	if err := pem.Encode(encoded, &pem.Block{Type: "CERTIFICATE", Bytes: certBytes}); err != nil {
+		return "", fmt.Errorf("unable to pem encode test certificate: %+v", err)
+	}
+
+	return encoded.String(), nil
+}
+
+func testCheckAzureRMAttestationDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Attestation.ProviderClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_attestation" {
+			continue
+		}
+		id, err := parse.AttestationId(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Attestation.ProviderClient: %+v", err)
+			}
+		}
+		return nil
+	}
+	return nil
+}
+
+// currently only supported in "East US 2", "West Central US" & "UK South"
+func testAccAzureRMAttestation_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-attestation-%d"
+  location = "%s"
+}
+`, data.RandomInteger, "uksouth")
+}
+
+func testAccAzureRMAttestation_basic(data acceptance.TestData) string {
+	template := testAccAzureRMAttestation_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_attestation" "test" {
+  name                = "ap%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMAttestation_update(data acceptance.TestData) string {
+	template := testAccAzureRMAttestation_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_attestation" "test" {
+  name                = "ap%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  tags = {
+    ENV = "Test"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMAttestation_requiresImport(data acceptance.TestData) string {
+	config := testAccAzureRMAttestation_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_attestation" "import" {
+  name                = azurerm_attestation.test.name
+  resource_group_name = azurerm_attestation.test.resource_group_name
+  location            = azurerm_attestation.test.location
+}
+`, config)
+}
+
+func testAccAzureRMAttestation_completeString(data acceptance.TestData, testCertificate string) string {
+	template := testAccAzureRMAttestation_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_attestation" "test" {
+  name                = "ap%d"
```

Comment:
can we add an `acctest` prefix here:

```suggestion
  name                = "acctestap%d"
```

Reply:
I originally had it that way but the naming rules for this resource are 3-24 length, lower case letters and numbers only... I will switch it to use a random number of fixed length to see if that will work.

Reply:
fixed